crates/node-core/src/init.rs

```diff
@@ -190,17 +192,15 @@ pub fn insert_genesis_hashes<DB: Database>(
 }
 
 /// Inserts history indices for genesis accounts and storage.
-pub fn insert_genesis_history<DB: Database>(
+pub fn insert_genesis_history<'a, 'b, DB: Database>(
     provider: &DatabaseProviderRW<DB>,
-    genesis: &reth_primitives::Genesis,
+    alloc: impl Iterator<Item = (&'a Address, &'b GenesisAccount)> + Clone,
 ) -> ProviderResult<()> {
     let account_transitions =
-        genesis.alloc.keys().map(|addr| (*addr, vec![0])).collect::<BTreeMap<_, _>>();
+        alloc.clone().map(|(addr, _)| (*addr, vec![0])).collect::<BTreeMap<_, _>>();
```

Comment:
should we avoid cloning here by walking the iterator and filling the account and storage transitions?

Reply:
as you can see from the diff, the line below didn't change, it takes an `IntoIterator`. trying to keep diff small in this pr, so again, plead "out of scope"